libs/utils/src/hmpb/ballot_adjudication_reasons.ts

```diff
@@ -1,46 +1,50 @@
 import {
   AdjudicationReason,
   AdjudicationReasonInfo,
-  Contest,
   ContestOption,
   ContestOptionId,
   Contests,
   MarkStatus,
+  WriteInAreaStatus,
 } from '@votingworks/types';
 import { throwIllegalValue } from '@votingworks/basics';
 import { allContestOptions } from './all_contest_options';
 
 export interface Options {
-  optionMarkStatus: (option: ContestOption) => MarkStatus;
+  optionStatus: (option: ContestOption) => {
+    markStatus: MarkStatus;
+    writeInAreaStatus: WriteInAreaStatus;
+  };
 }
```

Comment:
In this PR, I maintain the same layer of abstraction we currently have - this method returns _all_ possible adjudication reasons, and the consumer decides which ones are important. 

But I'm thinking like we probably will need to change how this works, because the "all possible adjudication reasons" is ostensibly different from state to state, and the combinations and interactions may be too complicated. 

For now, it works because if we're ignoring marked write-ins then the write-in area status is always `Ignored` and it doesn't factor in to overvote adjudication.

I want to commit this if it makes sense, because of time limits, but as adjudication rules get more complicated may want to rethink layers here.